AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/AWSIncomingSubscriptionEventPublisher.swift

```diff
@@ -15,22 +15,41 @@ import Combine
 final class AWSIncomingSubscriptionEventPublisher: IncomingSubscriptionEventPublisher {
 
     private let asyncEvents: IncomingAsyncSubscriptionEventPublisher
-    private let mapper: IncomingAsyncSubscriptionEventToAnyModelMapper
-
-    var publisher: AnyPublisher<MutationSync<AnyModel>, DataStoreError> {
-        mapper.publisher
+    private var mapper: IncomingAsyncSubscriptionEventToAnyModelMapper?
+    private let subscriptionEventSubject: PassthroughSubject<IncomingSubscriptionEventPublisherEvent, DataStoreError>
+    private var mapperSink: AnyCancellable?
+    var publisher: AnyPublisher<IncomingSubscriptionEventPublisherEvent, DataStoreError> {
+        return subscriptionEventSubject.eraseToAnyPublisher()
     }
 
     init(modelType: Model.Type, api: APICategoryGraphQLBehavior) {
+        self.subscriptionEventSubject = PassthroughSubject<IncomingSubscriptionEventPublisherEvent, DataStoreError>()
         self.asyncEvents = IncomingAsyncSubscriptionEventPublisher(modelType: modelType,
                                                                    api: api)
 
         let mapper = IncomingAsyncSubscriptionEventToAnyModelMapper()
         self.mapper = mapper
 
         asyncEvents.subscribe(subscriber: mapper)
+        self.mapperSink = mapper.publisher.sink(receiveCompletion: onReceiveCompletion(receiveCompletion:),
+                                                receiveValue: onReceive(receiveValue:))
+    }
+
+    private func onReceiveCompletion(receiveCompletion: Subscribers.Completion<DataStoreError>) {
+        subscriptionEventSubject.send(completion: receiveCompletion)
     }
 
+    private func onReceive(receiveValue: MutationSync<AnyModel>) {
+        subscriptionEventSubject.send(.mutationEvent(receiveValue))
+    }
+
+    func kill() {
+        mapperSink?.cancel()
```

Comment:
As above, I think our `kill` contract needs to be asynchronous, and emit a completion on its publisher when it is complete.